include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -547,7 +547,8 @@ struct __result_and_scratch_storage
     }
 
   public:
-    __result_and_scratch_storage(_ExecutionPolicy& __exec, ::std::size_t __scratch_n)
+    template <typename _Policy>
```

Comment:
Why does this policy need to be different to `_ExecutionPolicy`?

Reply:
I think this is more about being able to accept rvalue references to policies.  However, I'm not sure it is specifically necessary in any of the subsequent PRs.  

I think either we need to forward it into the `__exec` constructor, or accept a `const _ExecutionPolicy&`, which will enable all variety of argument.  

Also, I think we should rename the parameter to be distinct from the member variable.

Reply:
I propose to use a `const _ExecutionPolicy&` here instead.  Any copies required will happen once on entry to this constructor.  This should allow incoming rvalue policies, (or any other policy reference flavor).